### ISSUE TITLE
PublicApiAnalyzers fails with precise numeric constants

### ISSUE THREAD

**Issue by jpobst** [dotnet/roslyn-analyzers#6747]
### Analyzer

**Diagnostic ID**: [RS0016](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md): Add public types and members to the declared API

### Analyzer source

NuGet Package: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers/)

Version: 3.3.4 (Latest)

### Describe the bug

When using very precise `float` or `double` `const` values, the values are truncated when automatically added to `PublicAPI.Unshipped.txt` by the code fix.  This results in spurious warnings about public API.

### Steps To Reproduce

With these constants:

```csharp
public const double Pi = 3.1415926535897931;
public const float GravityDeathStarI = 3.53036143E-07f;
```

The code fix produces the following lines:

```
const public_analyzers_test.Class1.GravityDeathStarI = 3.5303614E-07 -> float
const public_analyzers_test.Class1.Pi = 3.141592653589793 -> double
```

These values do not match the code, so the following warnings are produced:

```
Class1.cs(7,24,7,41): warning RS0016: Symbol 'GravityDeathStarI' is not part of the declared public API
Class1.cs(6,25,6,27): warning RS0016: Symbol 'Pi' is not part of the declared public API
PublicAPI.Unshipped.txt(2,1,2,78): warning RS0017: Symbol 'const public_analyzers_test.Class1.GravityDeathStarI = 3.5303614E-07 -> float' is part of the declared API, but is either not public or could not be found
PublicAPI.Unshipped.txt(3,1,3,68): warning RS0017: Symbol 'const public_analyzers_test.Class1.Pi = 3.141592653589793 -> double' is part of the declared API, but is either not public or could not be found
```

**Comment by mavasani** [comment]
Investigated this further, and this seems to be a corner case where when the limits are reached for single precision and double precision numbers, `value.ToString()` truncates the last digit when executing in the IDE, but seems to keep the preserve it when running from command line build. I presume this is due to the CultureInfo differences between the execution hosts, but I am not sure.

Given that this is an extreme corner case, I am reluctant to file a compiler issue or spend too much time on investigation or a fix. If someone is interested to dig deeper and come up with a fix/root cause analysis, definitely go for it.
